core/src/test/java/org/apache/accumulo/core/client/rfile/RFileClientTest.java

```diff
@@ -840,4 +842,35 @@ public void testMultipleFilesAndCache() throws Exception {
     assertEquals(testData, toMap(scanner));
     scanner.close();
   }
+
+  @Test
+  public void testFileSystemFromUri() throws Exception {
+    String localFsClass = "LocalFileSystem";
+
+    String fileUri = "hdfs://127.0.0.5:8080/bulk-xyx/file1.rf";
+    // There was a bug in the code where the default hadoop file system was always used. This test
+    // checks that the hadoop filesystem used it based on the URI and not the default filesystem. In
+    // this env the default file system is the local hadoop file system.
+    var exception = assertThrows(IOException.class, () -> RFile.newWriter().to(fileUri).build());
```

Comment:
Tested this locally and it throws a `ConnectException.class`. 

If we wanted to be more specific we could capture the remoteFsHost address of `127.0.0.5:8080` and check the exception message for it.
```suggestion
    var exception = assertThrows(IOException.class, () -> RFile.newWriter().to(fileUri).build());
    var exception =
           assertThrows(ConnectException.class, () -> RFile.newWriter().to(fileUri).build());
    assertTrue(exception.getMessage().contains("to " + remoteFsHost
            + " failed on connection exception: java.net.ConnectException: Connection refused"));
```

Reply:
You'll need to add `import java.net.ConnectException;` for this suggestion.

I have these changes locally if you want me to push a commit.